src/components/legend/defaults.js

```diff
@@ -33,7 +33,14 @@ module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
         // *would* be shown by default, toward the two traces you need to
         // ensure the legend is shown by default, because this can still help
         // disambiguate.
-        if(trace.showlegend || trace._dfltShowLegend) {
+        if(trace.showlegend || (
+            trace._dfltShowLegend && !(
+                trace._module &&
+                trace._module.attributes &&
+                trace._module.attributes.showlegend &&
+                trace._module.attributes.showlegend.dflt === false
```

Comment:
Can you explain why we need this extra piece of logic:

```js
!(
                trace._module &&
                trace._module.attributes &&
                trace._module.attributes.showlegend &&
                trace._module.attributes.showlegend.dflt === false
```

here?

Reply:
Good question.
This is to keep the baselines to what was in place before.
Otherwise if there are e.g. one `scatter3d` with other gl3d traces (that now support legend) the `legend` would show up on the graph for `scatter3d`.  

![diff-gl3d_world-cals](https://user-images.githubusercontent.com/33888540/69650260-577b4500-103c-11ea-96a3-110fffcef6f4.png)